### DISCUSSION TITLE
Crashes with multiple tabs

### DISCUSSION THREAD

**Discussion by mikeyww** [qutebrowser/qutebrowser#5586] [Q&A] [answered]
The program crashes if more than a handful of tabs are opened at once.

**Comment by The-Compiler** [comment]
Not much I can say with so little information. Can you show your `:version` output? Can you reproduce with `--temp-basedir` as well?

**Comment by mikeyww** [comment]
https://paste.the-compiler.org/view/a519429b

**Comment by mikeyww** [comment]
The following Windows batch program will crash qutebrowser.
```
@echo off
TITLE Qutebrowser crash
REM -----------------------------------------------
REM This Windows batch program will crash qutebrowser.
REM -----------------------------------------------
SET n=0
SET browser="%ProgramW6432%\qutebrowser\qutebrowser.exe"
echo:
REM -----------------------------------------------
SET url[1]=http://www.amazon.com/
SET url[2]=http://www.dell.com/
SET url[3]=http://www.microsoft.com/
SET url[4]=http://www.hp.com/
SET url[5]=http://www.google.com/
SET url[6]=http://www.nbc.com/
SET url[7]=http://www.cbs.com/
SET url[8]=http://www.abc.com/
SET url[9]=http://www.aa.com/
SET url[10]=http://www.united.com/
SET url[11]=http://www.delta.com/
SET url[12]=http://www.epson.com/
SET url[13]=http://www.hp.com/
REM -----------------------------------------------
setlocal enabledelayedexpansion
REM -----------------------------------------------
:loop
SET /A n += 1
IF NOT DEFINED url[%n%] GOTO :done
echo %n%. Opening: !url[%n%]!
START "Browse" %browser% !url[%n%]!
TIMEOUT /T 1 >NUL
GOTO :loop
REM -----------------------------------------------
:done
echo:
Pause
EXIT
```

**Comment by The-Compiler** [comment]
Does `:set qt.force_software_rendering chromium` (and `:restart`) change anything?

**Reply by mikeyww** [reply]
This helped: the program did not crash the browser. Nonetheless, the browser did crash when more tabs were opened: I have a similar batch program that opens 37 pages. After reaching number 19, the browser [crashes](https://i.imgur.com/fzy5WBg.png).

> Failed to create OpenGL context for format
> 
> QSurfaceFormat(version 2-0, options
> QFlags<QSurfaceFormatz:FormatOption>(), depthBufferSize 24,
> redBufferSize -1, greenBufferSize -1, blueBufferSize -1,
> alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior
> QSurfaceFormat::DefaultSwapBehavior, swaplnterval 1,
> colorSpace QSurfaceFormat::DefaultColorSpace, profile
> QSurfaceFormat22NoProfile) .
> This is most likely caused by not having the necessary graphics
> drivers installed.
> 
> Install a driver providing OpenGL 2-0 or higher, or, if this is not
> possible, make sure the ANGLE Open GL ES 2.0 emulation
> libraries (libEGL.dll, libGLESv2-dll and d3dcompiler_*.dll) are
> available in the application executable's directory or in a location
> listed in PATH.

```
Operating system           : Microsoft Windows 7 Professional 
OS version                 : 6.1.7601
Service pack major version : 1
Service pack minor version : 0
Free physical memory       : 6687096
Total physical memory      : 17052119040
PC manufacturer            : TOSHIBA
PC model                   : PORTEGE R930
CPU                        : Intel64 Family 6 Model 58 Stepping 9
OS bitness                 : 64
CPU bitness                : 64
BIOS manufacturer          : TOSHIBA
BIOS name                  : Default System BIOS
BIOS version               : Version 6.90  
Motherboard manufacturer   : TOSHIBA
Motherboard product        : PORTEGE R930
Motherboard version        : Version A0
CPU cores                  : 2
Display width              : 1366
Display height             : 768
DPI                        : 120
Intel HD Graphics 4000
```

**Comment by The-Compiler** [comment, marked as answer]
Thanks for the additional information! This looks like a Windows-specific issue in Qt, see #4052. No idea what's going on exactly, though.
